draft-rescorla-tls-esni.md

```diff
@@ -606,8 +606,52 @@ lie. In this design, the attacker would need to be able to find a
 Z which would expand into a key that would validly AEAD-encrypt
 a message of his choice, which should be intractable (Hand-waving alert!).
 
+# Alternative SNI Protection Designs
 
-# Alternate Encryption Design
+Alternative approaches to encrypted SNI may be implemented at the TLS or 
+application layer. In this section we describe several alternatives and discuss 
+drawbacks in comparison to the design in this document.
+
+## TLS-layer
+
+### TLS in Early Data
+
+In this variant, real TLS Client Hellos are tunneled within the early data payload 
+of an outer TLS connection established with the client-facing server. This requires 
```

Comment:
Thank you for the changes.

Now I think I understand the approach suggested here; it's about encapsulating CH using 0-RTT data of an outer TLS connection, but sending the rest of the TLS transmissions directly (i.e. without double encryption).

Assuming that that is true, it might be worth clarifying that (e.g., state something like "Rest of the TLS messages / records are exchanged directly between the peers without double encryption").

Other than that, the PR looks to be ready for merge.

Reply:
@kazuho I added some clarifying text. Please have a look!